### PR TITLE
Fix alignment warnings in mcan

### LINKED ISSUE #41255: drivers/can/can_mcan.c: address-of-packed-member warnings
**Describe the bug**
Since cf111962e0d0c388bf806bbd9f17d49732daa264 warnings are generated on "address-of-packed-member"

3 of them are generated when compiling `drivers/can/can_mcan.c`
```
/local/mcu/zephyrproject/zephyr/drivers/can/can_mcan.c: In function 'can_mcan_init':
/local/mcu/zephyrproject/zephyr/drivers/can/can_mcan.c:389:2: error: converting a packed 'struct can_mcan_msg_sram' pointer (alignment 1) to a 'uint32_t' {aka 'unsigned int'} pointer (alignment 4) may result in an unaligned pointer value [-Werror=address-of-packed-member]
  389 |  for (uint32_t *ptr = (uint32_t *)msg_ram;
      |  ^~~
In file included from /local/mcu/zephyrproject/zephyr/drivers/can/can_mcan.c:11:
/local/mcu/zephyrproject/zephyr/drivers/can/can_mcan.h:148:8: note: defined here
  148 | struct can_mcan_msg_sram {
      |        ^~~~~~~~~~~~~~~~~
/local/mcu/zephyrproject/zephyr/drivers/can/can_mcan.c: In function 'can_mcan_get_message':
/local/mcu/zephyrproject/zephyr/drivers/can/can_mcan.c:525:15: error: taking address of packed member of 'struct can_mcan_rx_fifo' may result in an unaligned pointer value [-Werror=address-of-packed-member]
  525 |    for (src = fifo[get_idx].data_32,
      |               ^~~~
/local/mcu/zephyrproject/zephyr/drivers/can/can_mcan.c: In function 'can_mcan_send':
/local/mcu/zephyrproject/zephyr/drivers/can/can_mcan.c:699:9: error: taking address of packed member of 'struct can_mcan_tx_buffer' may result in an unaligned pointer value [-Werror=address-of-packed-member]
  699 |   dst = msg_ram->tx_buffer[put_idx].data_32,
      |         ^~~~~~~
```

**To Reproduce**
Could be reproduced with the following command:
```
twister -b -p nucleo_g474re -s tests/subsys/canbus/isotp/implementation/can.isotp.implemmentation -N
```

**Expected behavior**
No warnings at compile time

### PR DESCRIPTION
**Pull request by pdietl**

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
@carlescufi Please review as you are the author of  #40909.

File: drivers/can/can_mcan.c [hunk 1]
```diff
@@ -386,8 +388,8 @@ int can_mcan_init(const struct device *dev, const struct can_mcan_config *cfg,
 	}
 
 	/* No memset because only aligned ptr are allowed */
-	for (uint32_t *ptr = (uint32_t *)msg_ram;
-	     ptr < (uint32_t *)msg_ram +
+	for (uint32_t *ptr = (u32 *)msg_ram;
+	     ptr < (u32 *)msg_ram +
```
Comment (henrikbrixandersen) [inline drivers/can/can_mcan.c:392]: Could this be fixed by #41273 instead of adding `__may_alias`?

Comment (alexanderwachter) [reply drivers/can/can_mcan.c:392]: Yep. I commented in the pr you referring to.

Comment (pdietl) [reply drivers/can/can_mcan.c:392]: Yes, but you still reply on code that uses `__may_alias`, so it doesn't remove the usage.

**Comment by henrikbrixandersen** [comment]
Fixed by #41326. Closing.

**Pull request closed without merging**